### PR TITLE
Update for stringacolypse

### PR DESCRIPTION
**Pull request by malmaud**

### CODE REVIEW

File: src/Requests.jl [hunk 1]
```diff
@@ -86,7 +87,7 @@ get_request_settings() = SETTINGS
 for kind in [:Response, :Request]
     @eval bytes(r::$kind) = r.data
     @eval text(r::$kind) = utf8(bytes(r))
-    @eval Base.bytestring(r::$kind) = text(r)
+    @eval Base.String(r::$kind) = text(r)
```
Comment (TotalVerb) [inline src/Requests.jl]: not right for 0.3/0.4

File: src/Requests.jl [hunk 2]
```diff
@@ -122,7 +123,7 @@ function mimetype(r::Response)
         ct = split(headers(r)["Content-Type"], ";")[1]
         return Nullable(ct)
     else
-        return Nullable{UTF8String}()
+        return Nullable{String}()
```
Comment (TotalVerb) [inline src/Requests.jl]: `Compat.UTF8String` (`String` surprisingly aliases to `ByteString` on 0.3/0.4, not `UTF8String`)

File: src/Requests.jl [hunk 3]
```diff
@@ -136,7 +137,7 @@ function contentdisposition(r::Response)
             end
         end
     end
-    return Nullable{UTF8String}()
+    return Nullable{String}()
```
Comment (TotalVerb) [inline src/Requests.jl]: `Compat.UTF8String`

File: src/mimetypes.jl [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Derived from http://svn.apache.org/repos/asf/httpd/httpd/trunk/docs/conf/mime.types
 
-const MIMETYPES = Dict{UTF8String, UTF8String}()
+const MIMETYPES = Dict{String, String}()
```
Comment (TotalVerb) [inline src/mimetypes.jl]: `Compat.UTF8String`

File: src/multipart.jl [hunk 5]
```diff
@@ -30,9 +30,9 @@ write(io::ChunkedStream, arg) = write_chunked(io.io, arg)
 immutable FileParam
     file::Union{IO,Base.File,AbstractString,Vector{UInt8}}     # The file
     # The content type (default: "", which is interpreted as text/plain serverside)
-    ContentType::ASCIIString
-    name::ASCIIString                                  # The fieldname (in a form)
-    filename::ASCIIString                              # The filename (of the actual file)
+    ContentType::String
```
Comment (TotalVerb) [inline src/multipart.jl]: `Compat.ASCIIString`

File: src/streaming.jl [hunk 6]
```diff
@@ -7,7 +7,7 @@ type ResponseStream{T<:IO} <: IO
     buffer::IOBuffer
     parser::ResponseParser
     timeout::Float64
-    current_header::Nullable{ASCIIString}
+    current_header::Nullable{String}
```
Comment (TotalVerb) [inline src/streaming.jl]: `Compat.ASCIIString`

File: src/multipart.jl [hunk 7]
```diff
@@ -195,7 +195,7 @@ end
 
 immutable MultipartSettings
     datasizes::Vector{Int}
-    boundary::ASCIIString
+    boundary::String
```
Comment (TotalVerb) [inline src/multipart.jl]: `Compat.ASCIIString`

**Comment by TotalVerb** [comment]
Why no precompilation?

File: src/Requests.jl (hunk 1 above)
Comment (malmaud) [inline src/Requests.jl]: Thanks, you are quite right with all these comments.

**Comment by malmaud** [comment]
That automatically-generated precompile hints file wasn't actually speeding anything up, so it seemed prudent to remove it instead of continually revising it to cope with Julia updates.

**Comment by TotalVerb** [comment]
Makes sense. The package itself will still be precompiled though? That can save startup time.

**Comment by malmaud** [comment]
Yep.

**Comment by malmaud** [comment]
Closing in favor of https://github.com/JuliaWeb/Requests.jl/pull/127

**Pull request closed without merging**